rskj-core/src/test/java/co/rsk/peg/BridgeSupportRegisterBtcTransactionTest.java

```diff
@@ -0,0 +1,2553 @@
+            false
+        );
+
+        if (activations.isActive(ConsensusRule.RSKIP379)) {
+            when(provider.hasPegoutTxSigHash(firstInputSigHash)).thenReturn(true);
+        }
+
+        // act
+        BridgeSupport bridgeSupport = buildBridgeSupport(activations);
+        bridgeSupport.registerBtcTransaction(
+            rskTx,
+            btcTransaction.bitcoinSerialize(),
+            height,
+            pmt.bitcoinSerialize()
+        );
+
+        // assert
+        verify(provider, times(1)).setHeightBtcTxhashAlreadyProcessed(btcTransaction.getHash(false), rskExecutionBlock.getNumber());
+        Assertions.assertTrue(activeFederationUtxos.isEmpty());
+        Assertions.assertTrue(retiringFederationUtxos.isEmpty());
+    }
+
+    // Migration tests
+
+    @ParameterizedTest
+    @MethodSource("pre_and_post_rskip379_args")
+    void migration_ok(
+        ActivationConfig.ForBlock activations,
+        boolean shouldUsePegoutTxIndex
+    ) throws BlockStoreException, BridgeIllegalArgumentException, IOException {
+        // arrange
+        int height = shouldUsePegoutTxIndex ? blockNumberToStartUsingPegoutIndex : 1;
+
+        BtcTransaction btcTransaction = new BtcTransaction(btcMainnetParams);
+        Sha256Hash fundTxHash = BitcoinTestUtils.createHash(1);
+        btcTransaction.addInput(fundTxHash, FIRST_OUTPUT_INDEX, new Script(new byte[]{}));
+        btcTransaction.addOutput(Coin.COIN, activeFederation.getAddress());
+
+        Script p2SHScript = ScriptBuilder.createP2SHOutputScript(retiringFederation.getRedeemScript());
+        Script inputScript = p2SHScript.createEmptyInputScript(null, retiringFederation.getRedeemScript());
+        btcTransaction.getInput(FIRST_INPUT_INDEX).setScriptSig(inputScript);
+
+        FederationTestUtils.addSignatures(retiringFederation, retiringFedSigners, btcTransaction);
+
+        PartialMerkleTree pmt = createPmtAndMockBlockStore(btcTransaction, height);
+
+        when(provider.getOldFederation()).thenReturn(retiringFederation);
+
+        Sha256Hash firstInputSigHash = btcTransaction.hashForSignature(
+            FIRST_INPUT_INDEX,
+            retiringFederation.getRedeemScript(),
+            BtcTransaction.SigHash.ALL,
+            false
+        );
+
+        if (activations.isActive(ConsensusRule.RSKIP379)) {
+            when(provider.hasPegoutTxSigHash(firstInputSigHash)).thenReturn(true);
+        }
+
+        // act
+        BridgeSupport bridgeSupport = buildBridgeSupport(activations);
+        bridgeSupport.registerBtcTransaction(
+            rskTx,
+            btcTransaction.bitcoinSerialize(),
+            height,
+            pmt.bitcoinSerialize()
+        );
+
+        // assert
+        verify(provider, times(1)).setHeightBtcTxhashAlreadyProcessed(btcTransaction.getHash(false), rskExecutionBlock.getNumber());
+        Assertions.assertEquals(1, activeFederationUtxos.size());
+        Assertions.assertTrue(retiringFederationUtxos.isEmpty());
```

Comment:
Is it possible to assert that the `retiringFederationUtxos.size()` is 1 before calling ` bridgeSupport.registerBtcTransaction`?